reporting/CassandraBackend.cpp

```diff
@@ -573,49 +573,87 @@ CassandraBackend::fetchBookOffers(
     std::optional<ripple::uint256> const& cursor) const
 {
     CassandraStatement statement{selectBook_};
-    statement.bindBytes(book);
-    auto index = getIndexOfSeq(sequence);
-    if (!index)
-        return {};
-    BOOST_LOG_TRIVIAL(info) << __func__ << " index = " << std::to_string(*index)
-                            << " book = " << ripple::strHex(book);
-    statement.bindInt(*index);
-    if (cursor)
-        statement.bindBytes(*cursor);
-    else
-    {
-        ripple::uint256 zero = {};
-        statement.bindBytes(zero);
-    }
-    statement.bindUInt(limit);
-    CassandraResult result = executeSyncRead(statement);
 
-    BOOST_LOG_TRIVIAL(debug) << __func__ << " - got keys";
+    auto rng = fetchLedgerRange();
+    
+    if(!rng)
+        return {{},{}};
+
     std::vector<ripple::uint256> keys;
-    if (!result)
-        return {{}, {}};
-    do
+    uint32_t upper = sequence;
+    auto lastPage = rng->maxSequence - (rng->maxSequence % 256);
+
+    if (sequence != rng->minSequence)
     {
-        keys.push_back(result.getUInt256());
-    } while (result.nextRow());
+        upper = (sequence >> 8) << 8;
+        if (upper != sequence)
+            upper += (1 << 8);
```

Comment:
Use `BackendIndexer::getShift()` instead of hardcoding 256 and 8 here. Also, I'm adding a helper function that I'm going to push soon so you don't have to write this code here. It will be a member of `BackendInterface`. You can copy and paste it if you want:
```
    std::optional<uint32_t>
    getIndexOfSeq(uint32_t seq) const
    {
        if (!fetchLedgerRange())
            return {};
        if (fetchLedgerRange()->minSequence == seq)
            return seq;
        uint32_t shift = indexer_.getShift();
        uint32_t incr = (1 << shift);
        if ((seq % incr) == 0)
            return seq;
        return (seq >> shift << shift) + incr;
    }

```
Or just wait until I push this.